scripts/feature_tests/upgrade/upgrade_common.sh

```diff
@@ -209,12 +209,9 @@ function controlplane_has_correct_replicas() {
     replicas="${1}"
     echo "Waiting for all replicas of controlplane node"
     for i in {1..3600}; do
-        kubectl get secrets "${CLUSTER_NAME}"-kubeconfig -n "${NAMESPACE}" -o json | \
-          jq -r '.data.value'| base64 -d > /tmp/kubeconfig-"${CLUSTER_NAME}".yaml
-        cp_replicas=$(kubectl --kubeconfig=/tmp/kubeconfig-"${CLUSTER_NAME}".yaml get nodes |
-            awk 'NR>1' | grep -c master)
+        cp_replicas=$(kubectl get bmh -n metal3 | grep -i provisioned | grep -c controlplane)
```

Comment:
this means a change in the functionality of the test. bmh provisioned != node joined in target cluster. as it was before, it was checking that the node actually joined the cluster. by doing this change, you would miss all the errors that might happen once the node is booted but not joined yet (package installation error, kubeadm error etc.) I would not change this condition.

Reply:
change reverted